src/Microsoft.ML.AutoML/API/ExperimentBase.cs

```diff
@@ -102,19 +114,28 @@ public abstract class ExperimentBase<TMetrics, TExperimentSettings>
             const int crossValRowCountThreshold = 15000;
 
             var rowCount = DatasetDimensionsUtil.CountRows(trainData, crossValRowCountThreshold);
+            var samplingKeyColumnName = GetSamplingKey(columnInformation?.GroupIdColumnName, columnInformation?.SamplingKeyColumnName);
             if (rowCount < crossValRowCountThreshold)
             {
                 const int numCrossValFolds = 10;
-                var splitResult = SplitUtil.CrossValSplit(Context, trainData, numCrossValFolds, columnInformation?.SamplingKeyColumnName);
+                var splitResult = SplitUtil.CrossValSplit(Context, trainData, numCrossValFolds, samplingKeyColumnName);
                 return ExecuteCrossValSummary(splitResult.trainDatasets, columnInformation, splitResult.validationDatasets, preFeaturizer, progressHandler);
             }
             else
             {
-                var splitResult = SplitUtil.TrainValidateSplit(Context, trainData, columnInformation?.SamplingKeyColumnName);
+                var splitResult = SplitUtil.TrainValidateSplit(Context, trainData, samplingKeyColumnName);
                 return ExecuteTrainValidate(splitResult.trainData, columnInformation, splitResult.validationData, preFeaturizer, progressHandler);
             }
         }
 
+        private string GetSamplingKey(string groupIdColumnName, string samplingKeyColumnName)
+        {
+            UserInputValidationUtil.ValidateSamplingKey(samplingKeyColumnName, groupIdColumnName, _task);
+            if ( _task == TaskKind.Ranking)
+                return groupIdColumnName ?? DefaultColumnNames.GroupId;
+            return samplingKeyColumnName;
+        }
+
```

Comment:
Note: https://github.com/dotnet/machinelearning/blob/b47d3536f8c00b1930d681b15b6a0e653ceaa1fc/src/Microsoft.ML.AutoML/API/ExperimentBase.cs#L158 doesn't support Ranking, since there's no GroupId option.

Reply:
Can't you retrieve the GroupId from the metrics agent that is already saved inside the experiment MetricsAgent field?

---
In reply to: [452354163](https://github.com/dotnet/machinelearning/pull/5263#discussion_r452354163) [](ancestors = 452354163)

Reply:
As per discussion offline, the groupId will not be handled through the Metrics Agent. For this case, the groupId will be the default groupId name.